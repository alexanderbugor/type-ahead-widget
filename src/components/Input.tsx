import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import TextField from '@material-ui/core/TextField';

const useStyles = makeStyles({
  root: {
    width: '100%',
  },
});

type Props = {
  onChange: (event: React.ChangeEvent) => void,
  inputValue?: string,
};

const Input = ({ onChange, inputValue = '' }: Props) => {
  const classes = useStyles();

  return (
    <TextField
      value={inputValue}
      onChange={onChange}
      className={classes.root}
      id="outlined-basic"
      label="Search"
      variant="outlined"
    />
  );
};

export default Input;
