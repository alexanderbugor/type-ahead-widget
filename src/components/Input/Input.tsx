import React from 'react';
import { WithStyles, withStyles, createStyles } from '@material-ui/core/styles';
import TextField from '@material-ui/core/TextField';

const styles = createStyles({
  root: {
    width: '100%',
  },
});

interface Props extends WithStyles<typeof styles> {
  onChange: (event: React.ChangeEvent) => void,
  inputValue?: string,
}

export class Input extends React.PureComponent<Props> {
  componentDidMount() {
    document.addEventListener('keydown', this.handleKeydown, false);
  }

  handleKeydown = (event: KeyboardEvent) => {
    if (event.keyCode == 38 || event.keyCode == 40) {
      event.preventDefault();
    }
  }

  componentWillUnmount() {
    document.removeEventListener('keydown', this.handleKeydown, false);
  }

  render() {
    const { classes, inputValue, onChange } = this.props;

    return (
      <TextField
        value={inputValue || ''}
        onChange={onChange}
        className={classes.root}
        id="outlined-basic"
        label="Search"
        variant="outlined"
      />
    );
  }
}

export default withStyles(styles)(Input);
