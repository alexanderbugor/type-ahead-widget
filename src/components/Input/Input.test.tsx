import React from 'react';
import { TextField } from '@material-ui/core';
import { shallow } from 'enzyme';
import { Input } from './Input';

describe('Input', () => {
  const props = {
    onChange: jest.fn(),
    inputValue: 'MOCK_VALUE',
    classes: { root: 'MOCK_CLASS' },
  };

  it('should render and match snapshot', () => {
    const root = shallow<Input>(<Input {...props} />);

    expect(root).toMatchSnapshot();
  });

  describe('handleKeydown', () => {
    it('should prevent default event when up or down arrow is pressed', () => {
      const preventDefault = jest.fn();
      const root = shallow<Input>(<Input {...props} />);

      root.instance().handleKeydown({ keyCode: 38, preventDefault } as unknown as KeyboardEvent);
      root.instance().handleKeydown({ keyCode: 40, preventDefault } as unknown as KeyboardEvent);
      root.instance().handleKeydown({ keyCode: 42, preventDefault } as unknown as KeyboardEvent);

      expect(preventDefault).toHaveBeenCalledTimes(2);
    });
  });

  describe('onChange', () => {
    it('should be called by TextInput onChange', () => {
      const root = shallow<Input>(<Input {...props} />);
      const element = root.find(TextField);
      element.props().onChange('EVENT_MOCK' as unknown as React.ChangeEvent<HTMLInputElement>);

      expect(props.onChange).toHaveBeenCalledWith('EVENT_MOCK');
    });
  });
});
