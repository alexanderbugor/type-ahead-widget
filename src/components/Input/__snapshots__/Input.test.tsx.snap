// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render and match snapshot 1`] = `
<WithStyles(ForwardRef(TextField))
  className="MOCK_CLASS"
  id="outlined-basic"
  label="Search"
  onChange={[MockFunction]}
  value="MOCK_VALUE"
  variant="outlined"
/>
`;
